airflow/serialization/enums.py

```diff
@@ -44,6 +44,7 @@ class DagAttributeTypes(str, Enum):
     TIMEDELTA = "timedelta"
     TIMEZONE = "timezone"
     RELATIVEDELTA = "relativedelta"
+    BASE_TRIGGER = "base_trigger"
```

Comment:
How about just call this `trigger`?

Reply:
because there are two trigger classes. BaseTrigger is non-orm and Trigger is orm.

Reply:
BaseTrigger->Trigger is kindof like DAG->DagModel

Reply:
Ah OK. Do we need to be able to serialise `Trigger` too? (The naming is a bit confusing too fwiw, we use Foo and FooModel in most places instead, but this is not the time to change it.)

Reply:
The thing is, I feel we should call this `TRIGGER` regardless of the class name since we don’t need to have a type for the ORM class (at least I don’t think so? Do we?)

Reply:
Well... i put this to you.  Suppose we _did_ need a type for the ORM.  what would be your call in that case?

I think it's best to just assume we do need it and name given that assumption, even though not currently true.

Then if we ever do need it we won't have an issue.

I don't veto calling it the non-orm one `trigger`.  We could call it `trigger_non_orm`.  Or we could just plan to call the orm one `trigger_orm`.

Reply:
We can call it `trigger_model` like `dag_model`.

Thinking a bit, calling this `base_trigger` for now is probably not a bad idea as long as we don’t call the ORM class `trigger` (we can use `trigger_model` instead).

Reply:
ok. thanks.